client/task_runner.go

```diff
@@ -360,6 +363,9 @@ func (r *TaskRunner) run() {
 					panic("nil wait")
 				}
 
+				// Stop collection the task's resource usage
+				close(stopCollection)
```

Comment:
At this time, won't the `WaitCh()` unblock at the spot where we are listening for the event in the place where we are collecting resource usage? I had a different channel to stop collection in the first iteration, and removed it based on your comment and it felt right.